calculator.js

```diff
@@ -1,5 +1,146 @@
-const exampleAdditionInput = {
-  num1: 3,
-  num2: 5,
-  operation: 'add',
+
+const math_options = ['add', '+', 'subtract', '-', 'multiply', '*', 'divide', '/', 'exponentiation', '**', 'modulo', '%'];
+
+//TASKED WITH: The program should have support for these four operations: addition, subtraction, multiplication, and division.
+//TASKED WITH: The program should accept both the name and the symbol for each possible operation.
+const calculate = function calculate(input) {
+  confirmNumber(input);
+  confirmOperator(input);
+
+  if(input.operation === 'add' || input.operation === '+') {
+    return add(input);
+  }
+  else if(input.operation === 'subtract' || input.operation === '-')  {
+    return subtract(input);
+  }
+  else if(input.operation === 'multiply' || input.operation === '*') {
+    return multiply(input);
+  }
+  else if(input.operation === 'divide' || input.operation === '/') {
+    //TASKED WITH:The program handles divide when attempting to divide by zero
+    if(input.num2 === 0) {
+      console.log('Error: Cannot divide by 0.');
+    }
+    else {
+      return divide(input);
+    }
+  }
+  else if(input.operation === 'modulo' || input.operation === '%') {
+    return modulo(input);
+  }
 }
+
+//TASKED WITH: The program adds numbers accurately with both add and +
+const add = (input) => { console.log(exampleAdditionInput.num1 + exampleAdditionInput.num2); }
+
+//TASKED WITH:The program subtracts numbers accurately with both subtract and -
+const subtract = (input) => { console.log(exampleAdditionInput.num1 - exampleAdditionInput.num2); }
+
+//TASKED WITH:The program adds numbers accurately with both multiply and *
+const multiply = (input) => { console.log(exampleAdditionInput.num1 * exampleAdditionInput.num2); }
+
+//TASKED WITH:The program adds numbers accurately with both divide and /
+const divide = (input) => { console.log(exampleAdditionInput.num1 / exampleAdditionInput.num2); }
+
+//OPTIONAL ENHANCEMENT: Add support for the modulo operator (10 % 3 = 1).
+const modulo = (input) => { console.log(exampleAdditionInput.num1 % exampleAdditionInput.num2); }
+
+//TASKED WITH: The program also needs to handle erroneous operators
+//TASKED WITH:The program handles erroneous input. For example the user might enter clown when asked to enter a number
+
+const confirmOperator = function confirmOperator(input){
+  if(!math_options.includes(input.operation)) {
+    console.log('Error: Please provide a valid operation. \n Provided:', input.operation);
+  }
+}
+
+const confirmNumber = function confirmNumber(input) {
+  if(isNaN(input.num1)) {
+    console.log('Error: Please provide a valid number. \n Provided:', input.num1);
+  }
+  if(isNaN(input.num2)) {
+    console.log('Error: Please provide a valid number. \n Provided:', input.num2);
+  }
+}
+
+//################################
+//MANUAL TESTS BELOW FOR REFERENCE
+//################################
+
+// const exampleAdditionInput = {
```

Comment:
Great work providing thorough test cases to test the functionality of your program. Consider also outputting the entire mathematical expression to enhance user experience. You could write a helper function to do this.